File: tests/Mixed.java

public class Mixed {
	
	private int privateInstanceField;
	int instanceField;
	public int publicInstanceField;
	
	private static int privateStaticField;
	static int staticField;
	public static int publicStaticField;
	
	public static void testStatic(int value) {
		Mixed.privateStaticField = 4;
		Mixed.staticField = value;
		Mixed.publicStaticField = 2;
	}
	
	private static void testStatic2(Mixed mx) {
		mx.testInstance(Mixed.staticField);
		mx.privateInstanceField = Mixed.privateStaticField;
		mx.instanceField = Mixed.staticField;
		mx.publicInstanceField = Mixed.publicStaticField;
	}
	
	public void testInstance(int value) {
		this.privateInstanceField = 4;
		this.instanceField = value;
		this.privateInstanceField = 2;
	}
	
	public void testInstance2() {
		Mixed.privateStaticField = 0;
		Mixed.publicStaticField = 0;
		Mixed.staticField = 0;
		Mixed.testStatic2(this);
		this.testInstance(3);
	}
	
	public Mixed returnArg(Mixed self) {
		return self;
	}
	
	public void testMixed() {
		Mixed mx = new Mixed();
		mx.testInstance(42);
		mx = this.returnArg(new Mixed());
		Mixed.testStatic(42);
		Mixed.testStatic2(mx);
		
	}
}

File: translated_tests/1 Single class test/Mixed.c
#include"Mixed.h"

// instance fields of the class
struct Mixed {
	 int privateInstanceField;
	 int instanceField;
	 int publicInstanceField;
};

// static fields of the class
int static_privateStaticField;
int static_staticField;
int static_publicStaticField;

void testStatic(int value){
		static_privateStaticField = 4;
		static_staticField = value;
		static_publicStaticField = 2;
	}

void testStatic2(Mixed mx){
		testInstance(mx, static_staticField);
		mx.privateInstanceField = static_privateStaticField;
		mx.instanceField = static_staticField;
		mx.publicInstanceField = static_publicStaticField;
	}

void testInstance(Mixed this, int value){
		this.privateInstanceField = 4;
		this.instanceField = value;
		this.privateInstanceField = 2;
	}

void testInstance2(Mixed this){
		static_privateStaticField = 0;
		static_publicStaticField = 0;
		static_staticField = 0;
		testStatic2(this);
		testInstance(this, 3);
	}

Mixed returnArg(Mixed this, Mixed self){
		return self;
	}

void testMixed(Mixed this){
		Mixed mx = new_Mixed();
		testInstance(mx, 42);
		mx = returnArg(this, new_Mixed());
		testStatic(42);
		testStatic2(mx);
		
	}

